tempesta_fw/http_parser.c

```diff
@@ -462,30 +462,51 @@ parse_int_hex(unsigned char *data, size_t len, unsigned long *acc)
 }
 
 /**
- * Add unconditionaly hop-by-hop headers to list of hop-by-hop headers names.
- * Do not init raw headers table until we found that Connection header contains
- * atleast one raw header.
+ * Add spec header indexes to list of hop-by-hop headers
  */
-static void
-__hbh_parser_init(TfwHttpHbhHdrs *hbh_hdrs, bool req)
+static inline void
+__hbh_parser_init_req(TfwHttpReq *req)
+{
+	TfwHttpHbhHdrs *hbh_hdrs = &req->parser.hbh_parser;
+	/* Connection is hop-by-hop header by RFC 7230 6.1 */
+	hbh_hdrs->spec |= 0x1 << TFW_HTTP_HDR_CONNECTION;
```

Comment:
As this should be performed only once per message, shouldn't this be a direct assignment, not an OR operation?